client/lib/signup/step-actions/index.js

```diff
@@ -333,44 +335,17 @@ export function setThemeOnSite( callback, { siteSlug, themeSlugWithRepo } ) {
 		.catch( ( error ) => callback( [ error ] ) );
 }
 
-function getDIFMLiteCartItemFromDependencies( dependencies ) {
-	const {
-		newOrExistingSiteChoice,
-		selectedDesign,
-		selectedSiteCategory,
-		isLetUsChooseSelected,
-		siteTitle,
-		siteDescription,
-		twitterUrl,
-		facebookUrl,
-		linkedinUrl,
-		instagramUrl,
-		displayEmail,
-		displayPhone,
-		displayAddress,
-	} = dependencies;
-	const extra = {
-		selected_design: selectedDesign?.theme,
-		site_category: selectedSiteCategory,
-		new_or_existing_site_choice: newOrExistingSiteChoice,
-		let_us_choose_selected: isLetUsChooseSelected,
-		site_title: siteTitle,
-		site_description: siteDescription,
-		twitter_url: twitterUrl,
-		facebook_url: facebookUrl,
-		linkedin_url: linkedinUrl,
-		instagram_url: instagramUrl,
-		display_email: displayEmail,
-		display_phone: displayPhone,
-		display_address: displayAddress,
-	};
+function getDIFMLiteCartItemFromDependencies( dependencies, reduxStore ) {
+	const state = reduxStore.getState();
+	const difmState = getDIFMState( state );
+	const extra = createDIFMCartExtrasObject( difmState, dependencies );
 	const cartItem = { product_slug: WPCOM_DIFM_LITE, extra };
 	return cartItem;
 }
```

Comment:
I think this logic rightly belongs **inside** the `addDIFMLiteToCart` method below
https://github.com/Automattic/wp-calypso/blob/e852c8a39a886d3bb8fafa900e2fe47772fc5285/client/lib/signup/step-actions/index.js#L346-L358

It was a confusing indirection for me to go inside completely different method which has the main logic related to 
"Adding DIFM lite into the cart"

Reply:
The logic for "Adding DIFM lite into the cart" is still inside the `addDIFMLiteToCart` function. (Constructing `providedDependencies` and calling `processItemCart` with the required parameters)

The `getDIFMLiteCartItemFromDependencies` function does not add DIFM lite into the cart. Instead, it *builds the cart item*, which will be later added to the cart. I suppose this function can be named better to avoid confusion. How about `buildDIFMLiteCartItem`?

Reply:
Yeah that is what I meant, building is also part of the logic of adding DIFM lite to Cart I think. In my view all of the logic for the assembler and building the cart all need to come here. I have to dig into multiple modules to understand what is happening.

This is what I felt when reading the code it felt like it lacks cohesion because the essential parts are pushed away multiple levels away from where it should actually live. But yeah I guess it maybe a style preference on my part. Renaming the function can help.

Reply:
I based this change on your suggestion: https://github.com/Automattic/wp-calypso/pull/61896#discussion_r827628198

Since a selector was not appropriate here, I used an assembler instead.

Reply:
I did not envision this going into another level of indirection 😬 
Anyways let's go with this, we can probably iterate if this introduces friction.